server/controllers/mealController.js

```
@@ -1,116 +1,192 @@
-        message: 'meal added',
-        meal
+    })
+      .then((existingMeal) => {
+        if (existingMeal) {
+          return res.status(400)
+            .json({
+              status: 'fail',
+              message: 'this meal already exists'
+            });
+        }
+        return Meal.create({
+          category,
+          name,
+          price,
+          image,
+          userId: req.user.id
+        })
+          .then((newMeal) => {
+            const mealDetails = pick(newMeal, ['category', 'name', 'price', 'image']);
+            res.status(201)
+              .json({
+                status: 'success',
+                message: 'Meal created successfully',
+                meal: mealDetails
+              });
+          });
       });
   }
   /**
-   * PUT a new meal
-   * @param {any} req
-   * @param {any} res
-   * @returns {json} updates meal
-   * @memberof Meal
+   * @description - get all meals
+   * @static
+   *
+   * @param {object} req - HTTP Request
+   * @param {object} res - HTTP Response
+   *
+   * @memberof mealController
+   *
+   * @returns {object} Class instance
    */
-  updateMeal(req, res) {
-    const { id } = req.params;
-    let putMeal;
-    mealsDb.forEach((meal) => {
-      if (req.body.name === meal.name) {
-        return res.status(409).json({
-          status: 'fail',
-          message: 'meal name already exists, add another meal'
-        });
-      }
-      if (meal.id === parseInt(id, 10)) {
-        meal.category = req.body.category || meal.category;
-        meal.name = req.body.name || meal.name;
-        meal.price = req.body.price || meal.location;
-        putMeal = meal;
-      }
-    });
-    if (putMeal) {
-      return res.status(200).json({ status: 'successfully updated', message: 'meal updated', meal: putMeal });
+  static getAllMeals(req, res) {
+    const userRole = pick(req.user, ['role']);
+    if (userRole.role !== 'admin') {
+      return res.status(401).send();
```

You should also return an error message in your response. Not just the error code. Also `401` is not the appropriate code to send in this scenario